content/acra/configuring-maintaining/general-configuration/acra-server/encryptor-config.md

```diff
@@ -0,0 +1,450 @@
+
+So, it's important to declare correct order of columns due to it is only one way how to process queries like 
+`SELECT * FROM table1` or `INSERT INTO table1 VALUES (...)`. In first query it helps to understand how much and 
+which columns expect from the database, how to expand `*` value. In the second query it helps support `INSERT` command without explicitly 
+declared columns due to omitted section between table name `table1` and `VALUES` as it could be 
+`INSERT INTO table1 (column1, column2) VALUES(<value1>, <value2>)` with explicitly declared columns and order. 
+
+Without declared columns AcraServer **will not** support these kinds of queries.
+
+### **encrypted**
+
+This section declares security controls like `encryption`, `searchable encryption`, `masking`, `tokenization` on per-column basis for specified
+table.
+
+```
+schemas:
+- table: <...>
+  columns:
+    - data_column
+  encrypted:
+    - column: data_column
+
+      # General options
+      client_id: "<string>" # [optional] [conflicts_with=zone_id]
+      zone_id: "<string>" # [optional] [conflicts_with=client_id]
+      crypto_envelope: "<acrablock|acrastruct>" # [optional]
+      data_type: "<str|bytes|int32|int64>" # [optional] [conflicts_with=token_type|tokenized|consistent_tokenization]
+      default_data_value: "<string value>" # [optional] [required_with=data_type] may be string literal or valid int32/int64 yaml values
+
+      # Tokenization
+      token_type: "<int64|int32|str|bytes|email>" # [optional] [required_with=tokenized | consistent_tokenization]
+      tokenized: true # [optional] [default=false] [required_with=token_type]
+      consistent_tokenization: true # [optional] [default=false] [required_with=token_type]
+
+      # Masking
+      masking: "xxxx" # [optional] [required_with=plaintext_length | plaintext_side]
+      plaintext_length: 9  # [optional] [required_with=masking | plaintext_side]
+      plaintext_side: "<right|left>" # [optional] [required_with=masking | plaintext_side]
+
+      # Searching
+      searchable: true # [optional] [default=false]
+```
+
+Required: `true`
+
+Type: `array of items`
+
+Description: defines which security controls apply on column's data. There is can be configured [transparent encryption](/acra/security-controls/encryption/), 
+[masking](/acra/security-controls/masking/), [searchable encryption](/acra/security-controls/searchable-encryption/), 
+[tokenization](/acra/security-controls/tokenization/).
+
+This section allows to configure several groups of settings:
+- **common** 
+- **encryption**
+- **searchable encryption**
+- **masking**
+- **tokenization**
+
+<!-- TODO maybe it hard to understand and should be rephrased. -->
+Options from `common` group can be used together with options from other groups. But options from different groups 
+cannot be used together except `common`. Several can be used for several groups. One column can be configured only with 
+options from the one group and from `common`. 
```

Comment:
yeah pls try to rewrite

Reply:
any suggestions? TODO left for review to ask reviewers help with it)

Reply:
anyway, updated with examples of groups and cases